pkg/cloud-init/cloud-init.go

```diff
@@ -521,10 +521,10 @@ func GenerateLocalData(vmiName string, namespace string, data *CloudInitData) er
 		networkData = []byte(data.NetworkData)
 	}
 
-	diskutils.RemoveFile(userFile)
-	diskutils.RemoveFile(metaFile)
-	diskutils.RemoveFile(networkFile)
-	diskutils.RemoveFile(isoStaging)
+	diskutils.RemoveAllAndLogFailure(userFile)
+	diskutils.RemoveAllAndLogFailure(metaFile)
+	diskutils.RemoveAllAndLogFailure(networkFile)
+	diskutils.RemoveAllAndLogFailure(isoStaging)
 
```

Comment:
Some of the comments are not directly tied to the specific changes in the PR as I think the whole logic of this function (especially with the deleting/writing/cleaning of the files) should be re-visit.  The first thing would be to define what is considered as error and what is the consequence. 
1) For example, if the "Write()" fails for any of the files - the overall operation is aborted with an error, however if we fail to remove existing file we  do nothing - I would assume you want to abort in that case as well. 
2) The create/delete are not symmetrical which is always a potential issue - the code uses MkdirAll  which creates intermediate parts if they are missing, however there is no corresponding delete. So we are at risk of leaving some directory structure around
3) There are several error returns that don't cleanup files that were created up to that point. (e.g., os.Rename(isoStaging...)  if failed the isoStaging is not removed. 
4)  Not sure we need to use diskutils.RemoveAllAndLogFailure at all, why don't we use os.Remove() directly?  (I also don't understand what that function uses RemoveAll()) 
Maybe just define a removeExistingLocalFiles([]files)  and call it before and after the writes. 

BTW, does the files variable is used at all today?  Maybe I missed where ...

Reply:
@ezrasilvera as you said, the focus of my PR is the helper function, not `cloudinit.GenerateLocalData`. I would prefer to improve the function elsewhere, and filed pull #4928 for your comment about ineffectual `files` variable.